pkg/controller/podautoscaler/metrics/rest_metrics_client.go

```diff
@@ -58,11 +58,11 @@ type resourceMetricsClient struct {
 func (c *resourceMetricsClient) GetResourceMetric(resource v1.ResourceName, namespace string, selector labels.Selector) (PodMetricsInfo, time.Time, error) {
 	metrics, err := c.client.PodMetricses(namespace).List(metav1.ListOptions{LabelSelector: selector.String()})
 	if err != nil {
-		return nil, time.Time{}, fmt.Errorf("unable to fetch metrics from API: %v", err)
+		return nil, time.Time{}, fmt.Errorf("unable to fetch metrics from resource metrics API: %v", err)
 	}
 
 	if len(metrics.Items) == 0 {
-		return nil, time.Time{}, fmt.Errorf("no metrics returned from heapster")
+		return nil, time.Time{}, fmt.Errorf("no heapster generated metrics returned from resource metrics API")
```

Comment:
Why add the 'heapster generated' part? It's an implementation detail even if those metrics were actually generated by heapster and AFAIK they're not (I believe metrics-server scrapes them directly from kubelet using summary API).

Reply:
yeah, they come directly from metrics-server in this case, but better to just say "no metrics returned from resource metrics API"

Reply:
ditto for the other messages.

Reply:
Great point, will update accordingly! Thanks for the feedback :)